# 9-1 panic!과 함께하는 복구 불가능한 에러

`panic!` 매크로가 실행되면 실패메세지를 출력하고, 스택을 되감고 청소하고, 그 후 종료됩니다. 이런 일이 발생하는 가장 흔한 상황은 어떤 종류의 버그가 발견되었고 프로그래머가 이 에러를 어떻게 처리할지 명확하지 않을 때 입니다.

단순한 프로그램 내에서 `panic!`을 호출하면?

```rust
fn main() {
    panic!("crash and burn");
}
```
